### PR TITLE
Read into local hg repo

### LINKED ISSUE #1: Make HGFileInfo read info directly from the Mercurial repository
We can use the [python-hglib](https://www.mercurial-scm.org/wiki/PythonHglib) Python library.

### PR DESCRIPTION
**Pull request by calixteman**
Fixes #1.

### CODE REVIEW

**Comment by marco-c** [comment]
Have you considered using `python-hglib` instead of `mercurial`? It is suggested here: https://www.mercurial-scm.org/wiki/MercurialApi#Why_you_shouldn.27t_use_Mercurial.27s_internal_API.

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,28 @@
 language: python
-python:
-  - "2.7"
-  - "3.5"
+env:
+    # Enable python 2 and python 3 builds
+    - PYTHON=python2.7 PIP=pip2.7 PYSUF='2.7'
+    - PYTHON=python3 PIP=pip3 PYSUF=3
```
Comment (marco-c) [inline .travis.yml]: Why this change? To add the `PYSUF` env variable?
If this is the reason, you could use `TRAVIS_PYTHON_VERSION` instead, which is defined automatically by Travis.

File: tests/test_hgmozilla.py [hunk 2]
```diff
@@ -2,9 +2,75 @@
...
+            with open(myfile2, 'a') as In:
+                In.write(str(i))
+            commands.commit(ui, repo, myfile1, myfile2, message='message' + str(i), user='scooper@tbbt.com', addremove=True)
+            commands.push(ui, repo, dest=srcdir)
+            time.sleep(1.01)
+
+        return srcdir
+
+    def test_getfilelog(self):
+        if sys.version_info < (3, 0):
```
Comment (marco-c) [inline tests/test_hgmozilla.py]: Nit: to avoid having to indent the code after this, you can do:

```
if sys.version_info >= (3, 0):
    return

# Do stuff.
```

File: libmozdata/hgmozilla.py [hunk 3]
```diff
@@ -241,3 +246,41 @@ def get(paths, channel='nightly', node='tip'):
             FileInfo(queries=queries).wait()
 
         return data
+
+
+class HGMozilla(object):
```
Comment (marco-c) [inline libmozdata/hgmozilla.py]: Doesn't this class do something really similar to HGFileInfo? Perhaps we should move them together in the same file?
I wish we could have a single interface for them, so you don't have to write different code to use this one vs the other one.

File: .travis.yml (hunk 1 above)
Comment (marco-c) [inline .travis.yml]: OK, so yes, it isn't needed, you can simply use `TRAVIS_PYTHON_VERSION`.

File: libmozdata/hgmozilla.py (hunk 3 above)
Comment (calixteman) [inline libmozdata/hgmozilla.py]: Right now it's similar to HGFileInfo since we just have get_filelog, but we could add other stuff, so I'd prefer to keep it separately.

Comment (marco-c) [inline libmozdata/hgmozilla.py]: If we had the same interface, when we want to add new things, we could add them to both classes. This way the user of the library doesn't have to worry about which one they're using (the faster one or the slower one).

File: .travis.yml [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
+  - pip install --upgrade pip
+  - pip install "setuptools>=28.6.1"
   - pip install -r requirements.txt
   - pip install -r requirements-spikes.txt
   - pip install -r test-requirements.txt
 script:
   - flake8 .
   - coverage run --source=libmozdata -m unittest discover tests/
-  - python setup.py sdist
+  - $PYTHON setup.py sdist
```
Comment (marco-c) [inline .travis.yml]: This should be `python` again

File: .travis.yml [hunk 5]
```diff
@@ -1,17 +1,20 @@
 language: python
-python:
-  - "2.7"
-  - "3.5"
```
Comment (marco-c) [inline .travis.yml]: This should be re-added back

File: .travis.yml [hunk 6]
```diff
@@ -1,10 +1,15 @@
 language: python
-python:
```
Comment (marco-c) [inline .travis.yml]: This should be back too 😄

File: .travis.yml [hunk 7]
```diff
@@ -1,10 +1,15 @@
...
   - "2.7"
   - "3.5"
 before_install:
-  - sudo apt-get install -qq python-numpy python-scipy python3-numpy python3-scipy
```
Comment (marco-c) [inline .travis.yml]: These changes aren't needed, right? The old code worked

File: libmozdata/hgmozilla.py (hunk 3 above)
Comment (calixteman) [inline libmozdata/hgmozilla.py]: Sure, it was in my idea for a next step: just have a factory to choose the way to get info from hg repo and off course have the same interface.

File: .travis.yml (hunk 7 above)
Comment (calixteman) [inline .travis.yml]: yep it worked, but why install python3 pkgs when python 2 is used ?

Comment (marco-c) [inline .travis.yml]: OK, let's leave it like this.

File: .travis.yml [hunk 8]
```diff
@@ -3,8 +3,14 @@ python:
...
+  - if [[ ${TRAVIS_PYTHON_VERSION%%.*} == '3' ]]; then
+       sudo apt-get install -qq python3-numpy python3-scipy ;
+    else
+       sudo apt-get install -qq python-numpy python-scipy ;
+    fi
 install:
+  - pip install --upgrade pip
+  - pip install "setuptools>=28.6.1"
```
Comment (marco-c) [inline .travis.yml]: Could this be added to the requirements.txt file?
